### DISCUSSION TITLE
Can connect to Dart Observatory Connection closed - got an error: socket hang up

### DISCUSSION THREAD

**Discussion by pseudox07** [appium-userland/appium-flutter-driver#732] [Q&A]
Hi need help with the issue on Flutter Driver. After I start the Appium Inspector to check the connection or execute my test script from my python, the app was able to start, it stops the appium server and throw below error.

`[c4da9b13][FlutterDriver@83e0] Establishing a connection to the Dart Observatory
[c4da9b13][FlutterDriver@83e0] Connection to ws://127.0.0.1:39139/5GzSuz00HD8=/ws got an error: socket hang up
[c4da9b13][FlutterDriver@83e0] Deleting Flutter Driver session
[c4da9b13][FlutterDriver@83e0] Cleanup the port forward
[c4da9b13][AppiumDriver@1032] Event 'newSessionStarted' logged at 1725960522923 (17:28:42 GMT+0800 (Malaysia Time))
[c4da9b13][AppiumDriver@1032] Encountered internal error running command: Error: Cannot connect to the Dart Observatory URL ws://127.0.0.1:39139/5GzSuz00HD8=/ws. Check the server log for more details`

Below are my capabilities :
`{
  "platformName": "Android",
  "appium:platformVersion": "14.0",
  "appium:deviceName": "Pixel_8_API",
  "appium:udid": "emulator-5554",
  "appium:appPackage": "com.example.package.dev2",
  "appium:appActivity": "com.example.package.MainActivity",
  "appium:automationName": "Flutter",
  "appium:app": "C:/app/Accesibility ID/app-androidsit2-profile-2.apk",
  "appium:noReset": true
}`

already included the flutter_driver dart import & enableFlutterDriverExtension() in the main file
not sure where to start troubleshooting on this

**Comment by pseudox07** [comment]
> Hi need help with the issue on Flutter Driver. After I start the Appium Inspector to check the connection or execute my test script from my python, the app was able to start, it stops the appium server and throw below error.
> 
> `[c4da9b13][FlutterDriver@83e0] Establishing a connection to the Dart Observatory [c4da9b13][FlutterDriver@83e0] Connection to ws://127.0.0.1:39139/5GzSuz00HD8=/ws got an error: socket hang up [c4da9b13][FlutterDriver@83e0] Deleting Flutter Driver session [c4da9b13][FlutterDriver@83e0] Cleanup the port forward [c4da9b13][AppiumDriver@1032] Event 'newSessionStarted' logged at 1725960522923 (17:28:42 GMT+0800 (Malaysia Time)) [c4da9b13][AppiumDriver@1032] Encountered internal error running command: Error: Cannot connect to the Dart Observatory URL ws://127.0.0.1:39139/5GzSuz00HD8=/ws. Check the server log for more details`
> 
> Below are my capabilities : `{ "platformName": "Android", "appium:platformVersion": "14.0", "appium:deviceName": "Pixel_8_API", "appium:udid": "emulator-5554", "appium:appPackage": "com.example.package.dev2", "appium:appActivity": "com.example.package.MainActivity", "appium:automationName": "Flutter", "appium:app": "C:/app/Accesibility ID/app-androidsit2-profile-2.apk", "appium:noReset": true }`
> 
> already included the flutter_driver dart import & enableFlutterDriverExtension() in the main file not sure where to start troubleshooting on this

After restart my device and environment and run on real devices got this issue

"ext.flutter.driver" is not found in "extensionRPCs" ["ext.ui.window.impellerEnabled","ext.dart.io.httpEnableTimelineLogging","ext.dart.io.getSocketProfile","ext.dart.io.socketProfilingEnabled","ext.dart.io.clearSocketProfile","ext.dart.io.getVersion","ext.dart.io.getHttpProfile","ext.dart.io.getHttpProfileRequest","ext.dart.io.clearHttpProfile","ext.flutter.exit","ext.flutter.connectedVmServiceUri","ext.flutter.activeDevToolsServerAddress","ext.flutter.timeDilation","ext.flutter.profilePlatformChannels","ext.flutter.debugDumpRenderTree","ext.flutter.debugDumpSemanticsTreeInTraversalOrder","ext.flutter.debugDumpSemanticsTreeInInverseHitTestOrder","ext.flutter.profileRenderObjectPaints","ext.flutter.profileRenderObjectLayouts","ext.flutter.debugDumpApp","ext.flutter.debugDumpFocusTree","ext.flutter.showPerformanceOverlay","ext.flutter.didSendFirstFrameEvent","ext.flutter.didSendFirstFrameRasterizedEvent","ext.flutter.profileWidgetBuilds","ext.flutter.profileUserWidgetBuilds"]

**Reply by KazuCocoa** [reply]
It could indicate the package didn't include `flutter_driver:` package.
https://github.com/appium/appium-flutter-driver?tab=readme-ov-file#usage-and-requirement

**Reply by pseudox07** [reply]
For this issue, the flutter package has been included in the app build and no longer encounter this. But I did encounter the first issue back https://github.com/appium/appium-flutter-driver/discussions/732#discussion-7164053 and I believe this is similar  #728 . Let me know if you need additional info
